src/WebsocketClient/Program.cs

```diff
@@ -0,0 +1,370 @@
+            var optionWarmup = app.Option<int>("-w|--warmup <N>", "Duration of the warmup in seconds", CommandOptionType.SingleValue);
+            var optionDuration = app.Option<int>("-d|--duration <N>", "Duration of the test in seconds", CommandOptionType.SingleValue);
+            var optionScenario = app.Option("-s|--scenario <S>", "Scenario to run", CommandOptionType.SingleValue);
+            var optionLatency = app.Option<bool>("-l|--latency <B>", "Whether to collect detailed latency", CommandOptionType.SingleOrNoValue);
+
+            app.OnExecuteAsync(cancellationToken =>
+            {
+                Log("Websocket Client starting");
+
+                BenchmarksEventSource.Log.Metadata("websocket/client-version", "first", "first", "Client Version", "Client Version", "object");
+                BenchmarksEventSource.Measure("websocket/client-version", typeof(ClientWebSocket).GetTypeInfo().Assembly.GetCustomAttribute<AssemblyInformationalVersionAttribute>()?.InformationalVersion.ToString());
+
+                Scenario = optionScenario.Value();
+
+                ServerUrl = optionUrl.Value();
+
+                WarmupTimeSeconds = optionWarmup.HasValue()
+                    ? int.Parse(optionWarmup.Value())
+                    : 0;
+
+                ExecutionTimeSeconds = int.Parse(optionDuration.Value());
+
+                Connections = int.Parse(optionConnections.Value());
+
+                CollectLatency = optionLatency.HasValue()
+                    ? bool.Parse(optionLatency.Value())
+                    : false;
+
+                return RunAsync();
+            });
+
+            await app.ExecuteAsync(args);
+        }
+
+        private static async Task RunAsync()
+        {
+            await CreateConnections();
+
+            await StartScenario();
+
+            await StopJobAsync();
+        }
+
+        private static async Task StartScenario()
+        {
+            Log($"Starting scenario {Scenario}");
+
+            var tasks = new List<Task>(_connections.Count);
+
+            var cts = new CancellationTokenSource();
+            var tcs = new TaskCompletionSource<object>();
+            cts.Token.Register(() => tcs.SetResult(null));
+            cts.CancelAfter(TimeSpan.FromSeconds(ExecutionTimeSeconds));
+            _workTimer.Restart();
+
+            try
+            {
+                switch (Scenario)
+                {
+                    case "echo":
+                        for (var i = 0; i < _connections.Count; i++)
+                        {
+                            var id = i;
+                            // kick off a task per connection so they don't wait for other connections when sending "Echo"
+                            _ = Task.Run(async () =>
+                            {
+                                var buffer = new byte[1024 * 4];
+                                var message = Encoding.UTF8.GetBytes(DateTime.UtcNow.ToString("O"));
```

Comment:
I think this message should be configurable, either choose from a table of data (this could include images etc. for when we add compression tests), or just "send 1000 bytes" so we can test different payload sizes. For starters might be good to just choose a 100 or 1000 byte message of random bytes.

Reply:
I can change that, the question is what would be the best way to configure it.
1. I can add another CLI arg to set a number of bytes for a randomly generated message. The thing is that I'm not sure if this would be useful in comparison to some just constant size (let's say 1000).
2. I expect compression to be a different scenario, and that way it'd make more sense to have a different set of payloads for that (compression won't work well with random values). That's the reason I expect option 1 not to be reused for compression scenarios.

So my suggestion would be to do it simple for now - randomly generate a constant number of bytes for a message. Let me know what you think.

Reply:
> randomly generate a constant number of bytes for a message

Yes, that's what I was trying to get at.